src/cargo/sources/git/oxide.rs

```diff
@@ -0,0 +1,315 @@
+        }
+    }
+}
+
+/// Produce a repository with everything pre-configured according to `config`. Most notably this includes
+/// transport configuration. Knowing its `purpose` helps to optimize the way we open the repository.
+/// Use `config_overrides` to configure the new repository.
+pub fn open_repo(
+    repo_path: &std::path::Path,
+    config_overrides: Vec<BString>,
+    purpose: OpenMode,
+) -> Result<gix::Repository, gix::open::Error> {
+    gix::open_opts(repo_path, {
+        let mut opts = gix::open::Options::default();
+        opts.permissions.config = gix::permissions::Config::all();
+        opts.permissions.config.git_binary = purpose.needs_git_binary_config();
+        opts.with(gix::sec::Trust::Full)
+            .config_overrides(config_overrides)
+    })
+}
+
+/// Convert `git` related cargo configuration into the respective `git` configuration which can be
+/// used when opening new repositories.
+pub fn cargo_config_to_gitoxide_overrides(config: &Config) -> CargoResult<Vec<BString>> {
+    use gix::config::tree::{gitoxide, Core, Http, Key};
+    let timeout = HttpTimeout::new(config)?;
+    let http = config.http_config()?;
+
+    let mut values = vec![
+        gitoxide::Http::CONNECT_TIMEOUT.validated_assignment_fmt(&timeout.dur.as_millis())?,
+        Http::LOW_SPEED_LIMIT.validated_assignment_fmt(&timeout.low_speed_limit)?,
+        Http::LOW_SPEED_TIME.validated_assignment_fmt(&timeout.dur.as_secs())?,
+        // Assure we are not depending on committer information when updating refs after cloning.
+        Core::LOG_ALL_REF_UPDATES.validated_assignment_fmt(&false)?,
+    ];
+    if let Some(proxy) = &http.proxy {
+        values.push(Http::PROXY.validated_assignment_fmt(proxy)?);
+    }
+    if let Some(check_revoke) = http.check_revoke {
+        values.push(Http::SCHANNEL_CHECK_REVOKE.validated_assignment_fmt(&check_revoke)?);
+    }
+    if let Some(cainfo) = &http.cainfo {
+        values.push(
+            Http::SSL_CA_INFO.validated_assignment_fmt(&cainfo.resolve_path(config).display())?,
+        );
+    }
+    if let Some(user_agent) = &http.user_agent {
+        values.push(Http::USER_AGENT.validated_assignment_fmt(user_agent)?);
+    }
+    if let Some(ssl_version) = &http.ssl_version {
+        use crate::util::config::SslVersionConfig;
+        match ssl_version {
+            SslVersionConfig::Single(version) => {
+                values.push(Http::SSL_VERSION.validated_assignment_fmt(&version)?);
+            }
+            SslVersionConfig::Range(range) => {
+                values.push(
+                    gitoxide::Http::SSL_VERSION_MIN
+                        .validated_assignment_fmt(&range.min.as_deref().unwrap_or("default"))?,
+                );
+                values.push(
+                    gitoxide::Http::SSL_VERSION_MAX
+                        .validated_assignment_fmt(&range.max.as_deref().unwrap_or("default"))?,
+                );
+            }
+        }
```

Comment:
Should we continue the workaround on Windows here?

https://github.com/rust-lang/cargo/blob/39c13e67a5962466cc7253d41bc1099bbcb224c3/src/cargo/ops/registry.rs#L658-L674

Reply:
A good catch! It's probably better to continue with this restriction: c0354023d1baa6b13f6f4b4318687a82ab6ad5f9
 (I copied the text verbatim, with a note to indicate where to find the original).